tests/pre13/test_logic.py

```diff
@@ -211,6 +211,7 @@ def process(self, context):
     assert_equals(count["#"], 1)
 
 
+@with_setup(lib.setup_empty, lib.teardown)
```

Comment:
How come these decorators were added to the tests? Preferably they would only be assigned to those that need it. This test for example doesn't use the plug-in path at all, it's being told to run a specific plug-in directly.

Reply:
I'm a bit confused, most unit tests were failing without adding these decorators after my changes and I thought it was because these tests weren't protected against elements globally registered by other unit tests. 

But it must have been something wrong with my environment as I couldn't reproduce the error. I reverted these modifications...

Reply:
The decorator is allowing tests to affect global state, by reverting any change they make once finished. But the tests without it weren't affecting global state and thus doesn't need them. For readability, I would only add them where needed such that the reader knows that if there *is* a decorator, then the test does affect global state.